python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -646,41 +677,43 @@ cdef class Column:
         strides, size_type overflow) by a prior call to
         `_prepare_array_metadata`.
         """
-        if len(shape) == 1:
-            size = shape[0]
-            return Column(dtype, size, data, None, 0, 0, [])
-
-        else:
-            num_rows, num_cols = shape
+        ndim = len(shape)
+        flat_size = functools.reduce(operator.mul, shape)
+        data_col = Column(
+            data_type=dtype,
+            size=flat_size,
+            data=data,
+            mask=None,
+            null_count=0,
+            offset=0,
+            children=[],
+        )
 
-            offsets_col = sequence(
-                num_rows + 1,
-                Scalar.from_py(0, DataType(type_id.INT32)),
-                Scalar.from_py(num_cols, DataType(type_id.INT32)),
-            )
+        if ndim == 1:
```

Comment:
Nit: The `for i in range(ndim - 1, 0, -1)` loop below will just no-op for `ndim=1`, so this short-circut isn't _absolutely_ necessary